mesos-commons/src/main/java/org/apache/mesos/collections/Maps.java

```diff
@@ -0,0 +1,30 @@
+package org.apache.mesos.collections;
+
+import com.google.common.base.Predicate;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ */
+public class Maps {
+
+  public static Map<String, String> filterMapKeysThatStartWith(Map<String, String> map, final String startWithString) {
+    return com.google.common.collect.Maps.filterKeys(map, new Predicate<String>() {
+      @Override
+      public boolean apply(@Nonnull String s) {
+        return s.startsWith(startWithString);
+      }
+    });
+  }
+
```

Comment:
my only hesitation with `propertyMapFilter` is it isn't a generic filter, it is specifically a filter of start with.   I'll look at some options.